zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/ConditionalOnRestGatewayEnabled.java

```diff
@@ -23,7 +22,6 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.TYPE, ElementType.METHOD})
 @Documented
-@ConditionalOnWebApplication
```

Comment:
@Zelldon @nicpuppa Not sure whether this is 100% correct, but these configurations cannot be enabled when running non-webapp spring configurations. In case of the migration service, when running as a Standalone app, we do not need the whole web env setup.

Reply:
It's required to enable the _CamundaServices_ [configurations](https://github.com/camunda/camunda/blob/5709d730877b5e35cf8b3a0764f18ce1c4239416/dist/src/main/java/io/camunda/application/commons/service/CamundaServicesConfiguration.java#L150) required for Identity migration side at this moment at least. If they follow another strategy we can skip enabling these configuration beans.

Reply:
I see thanks, but I guess removing this might break also some parts 🤔 Or did it work?

Wondering how this works for the Standalone gateway 🤔

Reply:
It did work, and seems like it does not break anything :angel:  but not 100% sure that's why I asked for @nicpuppa 's input here.

AFAIK the properties inside the `@ConditionalOnProperty` are _AND_ bound and also the link between the two annotations is also _AND_.

Reply:
I'm not sure why I got pinged here @panagiotisgts. I think you mistakenly tag me instead @npepinpe (?)

Reply:
> I'm not sure why I got pinged here ...

Not once but twice... 

>Hm this would mean we cant run a gateway alone? As it is not a webapp? 🤔

It's run with the _Servlet_ application type.

Reply:
So the issue is that what, the migration app is not a web app (doesn't start a server), but we need some of the configurations annotated by `@ConditionalOnRestGatewayEnabled` in the migration app itself? It sounds like these should be pulled out with different conditions then, as they must now be shared?

I think I'm missing context here. I added this before we had these kind of requirements, so if you see a need to change things go ahead. Or did you have a question on how it works?